### PR TITLE
Avoid running lint jobs if `rs` files weren't changed

### PR DESCRIPTION
**Pull request by VitaliyaIoffe**
ci: avoid runs if rs files weren't changed

We don't need to run linting tasks if rust files are same with HEAD.
GithHub Actions provide ability to skip task by conditions. Was added
check that there are changed rust files. If check returns True we
execute linting and others workflows, otherwise, GitHub Actions
should skip them.

Close: pyrsia/pyrsia#479

## Runs links with example
when we have to skip jobs: [this](https://github.com/VitaliyaIoffe/pyrsia/actions/runs/2098578301)
when job should be triggered: [that](https://github.com/VitaliyaIoffe/pyrsia/actions/runs/2098526448)

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [X] I've requested a review  from `pyrsia/collaborators`.

### Code Contributions

I don't do next, because it is workflow changes

- [ ] I've built the code `cargo build --all-targets` successfully.
- [ ] I've run the unit tests `cargo test --workspace` and everything passes.

And I need some help to understand with what branches shoud we compare current. Is it good approach for your team?
Any other points?
Are these jobs chosen correctly?

### CODE REVIEW

**Comment by VitaliyaIoffe** [comment]
How can I add the pyrsia/collaborators to review?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,13 +1,40 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: check modified files
+        id: check_files
+        run: |
+            echo "Next files were changed"
+            git diff --name-only origin/main
```
Comment (VitaliyaIoffe) [inline .github/workflows/lint.yml:26]: Should change `origin/main` to target branch

Comment (sbtaylor15) [reply .github/workflows/lint.yml:26]: @VitaliyaIoffe - the tj-actions/changed-files@v18.1 action may be an easier way to go.  Check it out and see.  Either way it's up to you.

**Comment by sbtaylor15** [comment]
> How can I add the pyrsia/collaborators to review?

Do you have pyrsia/collaborators in the list when you select the Gear icon by the reviewers?

image

File: .github/workflows/lint.yml (hunk 1 above)
Comment (VitaliyaIoffe) [reply .github/workflows/lint.yml:26]: Hi @sbtaylor15, 
I'll check that action. I've seen it before but I wasn't confident that is ok for your project to bring an external action

**Comment by VitaliyaIoffe** [comment]
> > How can I add the pyrsia/collaborators to review?
> 
> Do you have pyrsia/collaborators in the list when you select the Gear icon by the reviewers?
> 

I see only next
image
It seems I can't add reviewers

**Comment by sbtaylor15** [comment]
> 

Yes, I think the issue is at our github org level.  I reached out to the org admins to check out the settings.  Keep a look out for a github invite.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (sbtaylor15) [reply .github/workflows/lint.yml:26]: Yes, we use external actions to help simplify the yaml.  Also, we have dependabot making sure we are on the latest version of the action.

**Comment by betarelease** [comment]
@VitaliyaIoffe I added them manually for you right now. But we will fix it in the permissions so that you can add them in the future. 

Thanks for your interest and PR!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (betarelease) [reply .github/workflows/lint.yml:26]: Yes external actions are fine. They don't affect the binary.

Comment (VitaliyaIoffe) [reply .github/workflows/lint.yml:26]: Ok, guys. I'll return with `tj-actions/changed-files@v18.1` variant. Thanks for your feedback

**Comment by VitaliyaIoffe** [comment]
Example when we skip jons: https://github.com/VitaliyaIoffe/pyrsia/actions/runs/2105219557
Else (I used yml pattern): https://github.com/VitaliyaIoffe/pyrsia/actions/runs/2105159416

**Comment by sbtaylor15** [comment]
@VitaliyaIoffe your changes look good.  Can you please remove the "Draft" from the PR?  Also, make sure the CLA is signed.

**Comment by VitaliyaIoffe** [comment]
> @VitaliyaIoffe your changes look good. Can you please remove the "Draft" from the PR? Also, make sure the CLA is signed.

done and done

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -7,7 +7,26 @@ permissions:
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get changed rust files
+        id: changed-rust-files
+        uses: tj-actions/changed-files@v18.6
+        with:
+          files: |
+            **/*.rs
```
Comment (tiainen) [inline .github/workflows/lint.yml:25]: We should at least also include all `Cargo.toml` and `Cargo.lock` files.

**Comment by VitaliyaIoffe** [comment]
Am I right that I also should add the same change for `coverage` job from `rust.yml`?

File: .github/workflows/lint.yml (hunk 2 above)
Comment (VitaliyaIoffe) [reply .github/workflows/lint.yml:25]: Added

**Review by tiainen — APPROVED** [review]

**Comment by prince-chrismc** [comment]
@erwin1 is off so @tiainen picked up this assigned PR (lots of vacations)

**Pull request merged**